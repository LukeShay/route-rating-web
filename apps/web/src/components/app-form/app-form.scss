:host {
  display: block;

  #form-box {
    box-sizing: border-box;

    *,
    *:before,
    *:after {
      box-sizing: inherit;
    }


    h1 {
      margin-top: 0;
    }


    margin: 2rem auto;
    width: 100%;
    max-width: 30rem;
    background: #fff;
    padding: 2rem;
    border-radius: 0.5rem;
  }

  rr-input {
    width: 100%;
    max-width: 600px;
    display: block;

    label {
      display: block;
      margin-bottom: 0.5rem;
    }

    input, textarea {
      box-sizing: border-box;
      -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
      outline: none;
      display: block;
      width: 100%;
      padding: 7px;
      border: none;
      border-bottom: 1px solid #ddd;
      background: transparent;
      margin-bottom: 10px;
      height: 45px;
      font-size: inherit;
      border-radius: 5px;
      border: 1px solid #ddd;
    }

    textarea {
      height: auto;
      font-family: inherit;
      resize: none;
    }

    select {
      width: 100%;
      border-radius: 5px;
      border: 1px solid #ddd;
      height: 45px;
      padding: 0 .5rem;
      background: url($select-menu-arrow) no-repeat right 1em center;
      -webkit-appearance: none;
      cursor: pointer;
      font-size: inherit;
      margin-bottom: 10px;
    }
  }
}
