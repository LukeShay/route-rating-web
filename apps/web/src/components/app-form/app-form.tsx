import { Component, Host, h } from '@stencil/core'

@Component({
  tag: 'app-form',
  styleUrl: 'app-form.scss',
})
export class AppForm {
  render() {
    return (
      <Host>
        <div id="form-box">
          <slot />
        </div>
      </Host>
    )
  }
}
