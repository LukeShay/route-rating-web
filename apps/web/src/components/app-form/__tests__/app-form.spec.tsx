import { AppForm } from '../app-form'
import { newSpecPage } from '@stencil/core/testing'

describe('app-form', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [AppForm],
      html: `<app-form></app-form>`,
    })
    expect(page.root).toMatchSnapshot()
  })
})
