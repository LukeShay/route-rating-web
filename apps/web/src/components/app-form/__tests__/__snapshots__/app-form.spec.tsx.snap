// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-form should match snapshot 1`] = `
<app-form>
  <!---->
  <div id="form-box"></div>
</app-form>
`;
